File: server/src/main/java/com/higgs/server/web/dto/ActionParameterDto.java
package com.higgs.server.web.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import javax.persistence.Transient;

@Data
public class ActionParameterDto {
    @NotNull
    private Long actionParameterSeq;

    private String name;

    private Long actionSeq;

    private ActionParameterTypeDto actionParameterType;

    private String defaultValue;

    @Transient
    @JsonInclude
    private String currentValue;
}

File: dotnet/HomeAssistantMainServer/Models/DTOs/ActionParameterDto.cs
namespace HomeAssistantMainServer.Models.DTOs;

public class ActionParameterDto
{
    public int ActionId { get; set; }
    public int ActionParameterTypeId { get; set; }
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string DefaultValue { get; set; } = string.Empty;
    public string CurrentValue { get; set; } = string.Empty;
}
